python/ray/dag/compiled_dag_node.py

```diff
@@ -150,54 +138,58 @@ def _wrap_exception(exc):
     return wrapped
 
 
-def _exec_task(self, task: "ExecutableTask", idx: int) -> bool:
+def _select_next_nodes(
+    actor_to_candidates: Dict["ray.actor.ActorHandle", List[DAGOperationGraphNode]],
+    graph: Dict[int, Dict[DAGNodeOperationType, DAGOperationGraphNode]],
+):
     """
-    Execute the task.
+    This function selects the next nodes for topological sort to generate execution
+    schedule. If there are multiple DAGOperationGraphNodes with zero in-degree,
+    select nodes based on the following rules:
+
+    #1  If the nodes are not NCCL write nodes, select the one with the smallest
+        `bind_index`. If there are multiple candidate nodes with the smallest
+        `bind_index` of the actors that they belong to, any one of them is
```

Comment:
"any of them is acceptable". 

Can you specify what deterministic order we apply here to break tie?

Reply:
`actor_to_candidates` is a dictionary. For Python 3.7, the Dictionary iteration order is guaranteed to be in order of insertion, so it should be the same order as the key of `actor_to_executable_tasks`.

Reply:
I think as long as when you call this API N times, if the order is guaranteed, that's fine. We can probably mention that instead.

Reply:
Added comments https://github.com/ray-project/ray/pull/46911/commits/bb6db7075c85f1a009f197b430155b91204fc04a

Reply:
Test: https://github.com/ray-project/ray/pull/46911/commits/fc123cbb610f198a117f722d65013ec752e0047c